### ISSUE TITLE
Feature Request: Resizing Area

### ISSUE THREAD

**Issue by egonelbre** [andlabs/ui#101]
I'm currently unable to figure out how to do an area that fills the entire window and the area size, including the handler, is updated when the window is resized. Am I missing something or is it currently not possible?

The use case is to create have a resizing canvas that is continously updated -- e.g. displaying a continuous graph. Here's the non-resizing example:

```
package main

import (
    "image"
    "image/color"
    "math/rand"
    "sync"
    "time"

    "github.com/andlabs/ui"
)

var (
    now      time.Time
    window   ui.Window
    graph    *Graph
    paintbox ui.Area
)

type Graph struct {
    sync.Mutex
    Values []int
    Canvas *image.RGBA

    Width  int
    Height int
}

func NewGraph() *Graph { return &Graph{} }

func (graph *Graph) SetSize(w, h int) {
    graph.Lock()
    graph.Unlock()

    graph.Width = w
    graph.Height = h
}

func (graph *Graph) Paint(rect image.Rectangle) *image.RGBA {
    graph.Lock()
    w, h := graph.Width, graph.Height

    cnv := graph.Canvas
    if cnv == nil || cnv.Bounds().Dx() != w || cnv.Bounds().Dy() != h {
        cnv = image.NewRGBA(image.Rect(0, 0, w, h))
        graph.Canvas = cnv
    }

    vals := graph.Values
    // I know that this is slow
    for x := 0; x < w; x++ {
        v := 0
        if x < len(vals) {
            v = vals[x]
        }
        for y := 0; y < h; y++ {
            if y < v {
                cnv.SetRGBA(x, y, color.RGBA{0, 0, 0, 0xFF})
            } else {
                cnv.SetRGBA(x, y, color.RGBA{0xFF, 0xFF, 0xFF, 0xFF})
            }
        }
    }
    graph.Unlock()

    return cnv.SubImage(rect).(*image.RGBA)
}

func (graph *Graph) AddValue(v int) {
    graph.Lock()
    defer graph.Unlock()

    if len(graph.Values) < graph.Width {
        graph.Values = append(graph.Values, v)
    } else {
        copy(graph.Values, graph.Values[1:])
        graph.Values = append(graph.Values[:graph.Width], v)
    }
}

func (graph *Graph) Mouse(me ui.MouseEvent)  {}
func (graph *Graph) Key(ke ui.KeyEvent) bool { return false }

var updater *time.Ticker

func feed() {
    updater = time.NewTicker(time.Millisecond)
    for range updater.C {
        graph.AddValue(rand.Intn(300))
        paintbox.RepaintAll()
    }
}

func main() {
    go ui.Do(func() {
        graph = NewGraph()
        graph.SetSize(320, 240)
        paintbox = ui.NewArea(320, 240, graph)
        window = ui.NewWindow("Graph", 640, 480, paintbox)

        go feed()
        window.OnClosing(func() bool {
            updater.Stop()
            ui.Stop()
            return true
        })
        window.Show()
    })

    err := ui.Go()
    if err != nil {
        panic(err)
    }
}
```

**Comment by andlabs** [comment]
I'm going to be redoing the Area system entirely; this shouldn't be an issue when that happens... Basically it'll be a canvas with virtual scrolling wher eyou define everything you need to draw.

Keep this open until then

**Comment by andlabs** [comment]
No longer relevant with new Area interface; also merged with andlabs/libui#329.
